components/script/dom/htmlareaelement.rs

```diff
@@ -9,13 +9,167 @@ use dom::bindings::js::{JS, MutNullableHeap, Root};
 use dom::bindings::str::DOMString;
 use dom::document::Document;
 use dom::domtokenlist::DOMTokenList;
+use dom::element::Element;
 use dom::htmlelement::HTMLElement;
 use dom::node::Node;
 use dom::virtualmethods::VirtualMethods;
+use euclid::point::Point2D;
 use html5ever_atoms::LocalName;
 use std::default::Default;
 use style::attr::AttrValue;
 
+#[derive(PartialEq)]
+#[derive(Debug)]
+pub enum Area {
+    Circle { left: f32, top: f32, radius: f32 },
+    Rectangle { top_left: (f32, f32), bottom_right: (f32, f32) },
+    Polygon { points: Vec<f32> },
+}
+
+pub enum Shape {
+    Circle,
+    Rectangle,
+    Polygon,
+}
+
+// https://html.spec.whatwg.org/multipage/#rules-for-parsing-a-list-of-floating-point-numbers
+impl Area {
+    pub fn parse(coord: &str, target: Shape) -> Option<Area> {
+        let mut array;
+        let size = coord.len();
+        let num = coord.as_bytes();
+        let mut index = 0;
+        let mut number_list = Vec::new();
+        let points_count;
+
+        match target {
+            Shape::Circle => points_count = 3,
+            Shape::Rectangle => points_count = 4,
+            Shape::Polygon => points_count = 0,
+        }
+
+        // Step 4: Walk till char is not a delimiter
+        while index < size {
+            let val = num[index];
+            match val {
+                0x2C | 0x3B | 0x20 | 0x09 | 0x0A | 0x0C | 0x0D  => {},
+                _ => break,
+            }
+
+            index += 1;
+        }
+
+        // Step 5: walk till end of string
+        while index < size {
+            array = Vec::new();
```

Comment:
I think it would be better to initialize it outside the loop, and `clear()` it inside each time (instead of reallocating a new one each time)